app/models/piece.rb

```diff
@@ -89,4 +106,58 @@ def create_move!(from_row, from_col)
       game_id: game.id
     )
   end
+
+  def move_obstructed_horizontal(to_col)
+    obstruction_array = []
+    col_direction = to_col > col ? 1 : -1
+    current_col = col + col_direction
+    while (to_col - current_col).abs > 0
+      obstruction_array << [row, current_col]
+      current_col += col_direction
+    end
+    Rails.logger.debug "Horizonal Move. Inspect #{obstruction_array.size} squares"
+    obstruction_array
+  end
+
+  def move_obstructed_vertical(to_row)
+    obstruction_array = []
+    row_direction = to_row > row ? 1 : -1
+    current_row = row + row_direction
+    while (to_row - current_row).abs > 0
+      obstruction_array << [current_row, col]
+      current_row += row_direction
+    end
+    Rails.logger.debug "Vertical Move. Inspect #{obstruction_array.size} squares"
+    obstruction_array
+  end
+
+  def move_obstructed_diagonal(to_row, to_col)
+    obstruction_array = []
+    col_direction = to_col > col ? 1 : -1
+    row_direction = to_row > row ? 1 : -1
+    current_col = col + col_direction
+    current_row = row + row_direction
+    while (to_row - current_row).abs > 0 && (to_col - current_col).abs > 0
+      obstruction_array << [current_row, current_col]
+      current_col += col_direction
+      current_row += row_direction
+    end
+    Rails.logger.debug "Diagonal Move. Inspect #{obstruction_array.size} squares"
+    obstruction_array
+  end
+
+  # moved capture piece methods to private - after verify that move to new square is valid
+  def capture_piece!(to_row, to_col)
+    captured_piece = game.piece_at(to_row, to_col)
+    captured_piece.update_attribute(:is_captured, true) if captured_piece
+  end
+
+  # This updates a piece to captured.
+  # MeO. This should be a private class based upon information hiding.
+  # However, cannot only call it on the instance of this class which is
+  # the piece being moved. Cannot call it on the other piece.
+  # Decided to move the update for other_piece into method capture_piece.
+  def captured!
+    update(is_captured: true)
+  end
 end
```

Comment:
Here is where my Ruby knowledge is lacking. I did see the declaration of Class Board within Class Game - but I never saw the creation of the object of Class Board. Is there an implicitly created instance of Board within the models. I now see that there is one in game_spec.rb.

I would much prefer using the grid array - rather than having to create an array of tuples and having to do a database query for each tuple. Let me look at game_spec.rb and consider a refactor of move_obstructed? I won't do anything until after our VTC.